articles/quickstart/backend/java-spring-security5/interactive.md

```diff
@@ -0,0 +1,150 @@
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.springframework.boot:spring-boot-starter-oauth2-resource-server'
+}
+```
+
+If you are using Maven, add the Spring dependencies to your `pom.xml` file:
+
+```xml
+// pom.xml
+
+<parent>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>2.5.12</version>
+    <relativePath/>
+</parent>
+
+<dependencies>
+    <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-web</artifactId>
+    </dependency>
+    <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
+    </dependency>
+</dependencies>
+```
+
+## Validate the audience {{{ data-action=code data-code="AudienceValidator.java#13:20" }}}
+
+To validate the JWT, you also need to validate that the JWT is intended for your API by checking the `aud` claim of the JWT. Create a new class named `AudienceValidator` that implements the `OAuth2TokenValidator` interface and override the `validate` method to verify whether the required `audience` is present.
+
+## Configure the resource server {{{ data-action=code data-code="SecurityConfig.java" }}}
+
+To configure the application as a Resource Server and validate the JWTs, create a class that extends `WebSecurityConfigurerAdapter`, add the `@EnableWebSecurity` annotation, and override the `configure` method
+
+### Protect API Endpoints
+
+<%= include('../_includes/_api_endpoints') %>
+
+The example below shows how to secure API methods using the `HttpSecurity` object provided in the `configure()` method of the `SecurityConfig` class. Route matchers are used to restrict access based on the level of authorization required:
+
+::: note
+By default, Spring Security will create a `GrantedAuthority` for each scope in the `scope` claim of the JWT. This is what enables using the `hasAuthority("SCOPE_read:messages")` method to restrict access to a valid JWT that contains the `read:messages` scope.
+
+If your use case requires different claims to make authorization decisions, see the [Spring Security Reference Documentation](https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#oauth2resourceserver-authorization-extraction) to learn how to customize the extracted authorities.
+:::
+
+### Configure JWT Validator
+
+Update the `SecurityConfig` class to configure a `JwtDecoder` bean that uses the `AudienceValidator`. The `auth0.audience` value from `application.yml` will be used to validate the `aud` claim. The `issuer` is verified using the `issuer-uri` value from the `application.yml`
+
+## Create the Domain Object {{{ data-action=code data-code="Message.java#1:11" }}}
+
+Create a new class named `Message`. This will act as a simple domain object to return during the API calls.
```

Comment:
We have removed this by avoiding creation of the POJO. Though the use of POJO returns a proper JSON, for the sake of sample, we will just return a String

Reply:
We cannot remove this as explained [here](https://github.com/auth0/docs/pull/10087#discussion_r883553502) so we have added more description to this section to avoid scroll issues